src/hooks/useUrlState.ts

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import { useHistory, useLocation } from 'react-router-dom';
+import { filtersFromQuery } from '../attribute-filters';
+import { AttributeList, Filters } from '../components/filters/filter.types';
+import { useQueryParams } from './useQueryParams';
+
+interface UseUrlState {
+  defaultQuery?: string;
+  attributes: AttributeList;
+}
+
+const QUERY_PARAM_KEY = 'q';
+const TENANT_PARAM_KEY = 'tenant';
+const SHOW_RESOURCES_PARAM_KEY = 'showResources';
+const DEFAULT_TENANT = 'application';
+const DEFAULT_SHOW_RESOURCES = '0';
+const DEFAULT_QUERY = '{ log_type =~ ".+" } | json';
+
+export const useUrlState = ({
```

Comment:
```suggestion
export const useURLState = ({
```
Nit: To match `setTenantInURL`, `setQueryInURL`, etc.